### PR TITLE
Make settings dynamic for MATERIALIZED_COLUMNS_ENABLED

### PR DESCRIPTION
**Pull request by fuziontech**
## Changes

We have a few settings that would be super convenient if they were environment variables _and_ could be dynamically set.

One such example is #7657 where we need to disable and pause materialized columns as we do a migration on the events table. 

Other places were we could leverage this is:
- email smtp / api settings
- Fine tune schedules for caching or aggregating
- Anything that is currently a setting could be ported

This code is pretty rough and could be improved a ton! Definitely a refactor in the future

## How did you test this code?

Run it and check out django admin console. Change the settings and watch from django shell as the setting flip flops.
image

## Will this work with multiple concurrent task instances?
Looks like it will since it fires off a query to the DB on `get` of the attribute
https://github.com/jazzband/django-constance/blob/master/constance/backends/database/__init__.py#L80

### CODE REVIEW

File: ee/clickhouse/models/test/test_property.py [hunk 1]
```diff
@@ -423,6 +423,7 @@ def test_prop_event_denormalized_ints(self):
         self.assertEqual(len(self._run_query(filter)), 1)
 
 
+@pytest.mark.django_db
```
Comment (yakkomajuri) [inline ee/clickhouse/models/test/test_property.py:426]: context?

**Review by yakkomajuri — APPROVED** [review]
i mean this looks good to me but haven't tested it

**Comment by yakkomajuri** [comment]
@fuziontech any further context on the failing test? would like to get this in

File: ee/clickhouse/models/test/test_property.py (hunk 1 above)
Comment (fuziontech) [reply ee/clickhouse/models/test/test_property.py:426]: This PR stores this setting in a django db model so the test needs access to the db to grab whether or not materialized columns are enabled!

**Comment by fuziontech** [comment]
>@fuziontech any further context on the failing test? would like to get this in

I have no further context on this one, but I can dig into it if I have time today to try to get to the bottom of it.

**Comment by fuziontech** [comment]
Ah! Got it. @EDsCODE helped me out. To update the snapshot definitions you have to add ` --snapshot-update` to `pytest` ✅

File: ee/clickhouse/models/test/test_property.py [hunk 2]
```diff
@@ -433,14 +434,21 @@ def test_parse_prop_clauses_defaults(snapshot):
         }
     )
 
-    assert parse_prop_clauses(filter.properties) == snapshot
+    assert parse_prop_clauses(filter.properties, allow_denormalized_props=False) == snapshot
```
Comment (macobo) [inline ee/clickhouse/models/test/test_property.py:437]: This test is no longer testing the correct thing though. Will need to fix this separately.

File: latest_migrations.manifest [hunk 3]
```diff
@@ -2,6 +2,7 @@ admin: 0003_logentry_add_action_flag_choices
 auth: 0012_alter_user_first_name_max_length
 axes: 0006_remove_accesslog_trusted
 contenttypes: 0002_remove_content_type_name
+database: 0002_auto_20190129_2304
```
Comment (macobo) [inline latest_migrations.manifest:5]: This shouldn't be here right?

File: ee/clickhouse/models/test/test_property.py (hunk 2 above)
Comment (EDsCODE) [reply ee/clickhouse/models/test/test_property.py:437]: Oh I see this is was meant to test defaults. Yeah the query started generating using materialized columns which didn't match preexisting snapshots

**Pull request merged**